src/config/artifacts.rs

```diff
@@ -0,0 +1,154 @@
+        return Err(OrandaError::Other(String::from(
+            "The repository and version are required for cargo_dist",
+        )));
+    }
+
+    let url = create_download_link(config, &String::from("dist-manifest.json"));
+
+    let resp = reqwest::blocking::get(url);
+
+    let Ok(resp) = resp else {
+        return Err(OrandaError::Other(String::from(
+            "The repository and version configurations are required for cargo_dist",
+        )));
+      };
+
+    let typed = &resp.json::<DistManifest>()?;
+
+    let mut html: Vec<Box<a<String>>> = vec![];
+    for release in typed.releases.iter() {
+        for artifact in release.artifacts.iter() {
+            if let ArtifactKind::ExecutableZip = artifact.kind {
+                let mut targets = String::new();
+                for targ in artifact.target_triples.iter() {
+                    targets.push_str(format!("{} ", targ).as_str());
+                }
+                let url = create_download_link(config, &artifact.name);
+
+                html.extend(
+                    html!(<a href=url class="block hidden" data-targets=targets><button class="business-button primary">{text!("Download")}</button></a>),
+                );
+            }
+        }
+    }
+
+    build_artifacts_html(config, typed)?;
+    Ok(Some(html!(
+    <div class="artifacts">
+        {html}
+        <a href="/artifacts.html" class="download-all">{text!("View all downloads")}</a>
+    </div>
+    )))
+}
+
+pub fn get_os_script(config: &Config) -> Result<Box<script<String>>> {
+    let detect_os_js = axoasset::copy("src/site/javascript/detect_os.js", &config.dist_dir);
+
+    let path = tokio::runtime::Handle::current().block_on(detect_os_js)?;
+    let path_as_string = path.strip_prefix(&config.dist_dir)?.to_string_lossy();
+    Ok(html!(<script src=path_as_string />))
+}
+
+// False positive duplicate allocation warning
+// https://github.com/rust-lang/rust-clippy/issues?q=is%3Aissue+redundant_allocation+sort%3Aupdated-desc
+#[allow(clippy::vec_box)]
+fn create_content(table: Vec<Box<tr<String>>>) -> Box<div<String>> {
+    html!(
+    <div>
+        <h1>{text!("All downloads")}</h1>
+        <table>
+            <tr>
+                <th>{text!("Name")}</th>
+                <th>{text!("Kind")}</th>
+                <th>{text!("Download")}</th>
+            </tr>
+            <tbody>
+                {table}
+            </tbody>
+        </table>
+    </div>
+    )
+}
+
+pub fn build_artifacts_html(config: &Config, manifest: &DistManifest) -> Result<()> {
+    let mut table = vec![];
+    for release in manifest.releases.iter() {
+        for artifact in release.artifacts.iter() {
+            let name = &artifact.name;
+            let url = create_download_link(config, name);
+            table.push(html!(
+            <tr>
+                // <td>{name}</td>
+                // <td>{artifact.kind}</td>
+                <td><a href=url>{text!("Download")}</a></td>
+            </tr>
```

Comment:
If I uncomment any of these lines I get this massive error

```
error[E0599]: the method `into_iter` exists for reference `&std::string::String`, but its trait bounds were not satisfied
   --> src/config/artifacts.rs:139:24
    |
139 |               table.push(html!(
    |  ________________________^
140 | |             <tr>
141 | |                  <td>{name}</td>
142 | |                 // <td>{artifact.kind}</td>
143 | |                 <td><a href=url>{text!("Download")}</a></td>
144 | |             </tr>
145 | |             ));
    | |_____________^
    |
   ::: /Users/saravieira/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/string.rs:367:1
    |
367 |   pub struct String {
    |   -----------------
    |   |
    |   doesn't satisfy `std::string::String: IntoIterator`
    |   doesn't satisfy `std::string::String: Iterator`
    |
    = note: the following trait bounds were not satisfied:
            `&std::string::String: Iterator`
            which is required by `&std::string::String: IntoIterator`
            `&&std::string::String: Iterator`
            which is required by `&&std::string::String: IntoIterator`
            `&mut &std::string::String: Iterator`
            which is required by `&mut &std::string::String: IntoIterator`
            `std::string::String: Iterator`
            which is required by `std::string::String: IntoIterator`
            `&mut std::string::String: Iterator`
            which is required by `&mut std::string::String: IntoIterator`
            `str: Sized`
            which is required by `str: IntoIterator`
            `str: Iterator`
            which is required by `str: IntoIterator`
            `&str: Iterator`
            which is required by `&str: IntoIterator`
            `&mut str: Iterator`
            which is required by `&mut str: IntoIterator`
    = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
This is a baffling error but I'm blind-guessing it's mad that `artifact.kind` doesn't implement Display -- for now you should match on it and choose the strings you want the different values to become.

Reply:
that was also my assumption but even leaving that one commented it errors out, I think it's a axohtml bug

Reply:
ended up using div's